metpy/calc/thermo.py

```diff
@@ -1444,3 +1444,61 @@ def mixed_layer(p, *args, **kwargs):
         ret.append((-1. / actual_depth.m) * np.trapz(datavar_layer, p_layer) *
                    datavar_layer.units)
     return ret
+
+
+@exporter.export
+@check_units('[pressure]', '[temperature]')
+def thickness_hydrostatic(pressure, temperature, **kwargs):
+    r"""Calculate the thickness of a layer via the hypsometric equation.
+
+    This thickness calculation uses the pressure and temperature profiles (and optionally
+    mixing ratio) via the hypsometric equation with virtual temperature adjustment
+
+    .. math:: \Delta z = -\frac{R_d}{g} \int_{p_0}^{p_1} T_v d\ln p
+
+    This assumes a hydrostatic atmosphere.
+
+    Layer bottom and depth specified in pressure.
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        Atmospheric pressure profile
+    temperature : `pint.Quantity`
+        Atmospheric temperature profile
+    mixing : `pint.Quantity`, optional
+        Profile of dimensionless mass mixing ratio. Defaults to zero, so that virtual
+        temperature is simply the temperature.
+    molecular_weight_ratio : `pint.Quantity` or float, optional
+        The ratio of the molecular weight of the constituent gas to that assumed
+        for air. Defaults to the ratio for water vapor to dry air.
+        (:math:`\epsilon\approx0.622`).
+    bottom: `pint.Quantity`, optional
+        The bottom of the layer in pressure. Default is the first observation, assumed
+        to be the surface.
+    depth: `pint.Quantity`, optional
+        The depth of the layer in hPa. Defaults to full range of given pressure values.
+
+    Returns
+    -------
+    `pint.Quantity`
+        The thickness of the layer in meters.
+
+    See Also
+    --------
+    pressure_to_height_std, virtual_temperature
+
+    """
+    mixing = kwargs.pop('mixing', np.zeros_like(temperature))
```

Comment:
Instead of calculating virtual temp with 0 mixing, could you instead do:
```python
mixing = kwargs.pop(‘mixing’, None)
if mixing is None:
    layer_virttemp = temperature
else:
    layer_virttemp = virtual_temperature(...)
```
I appreciate the elegance of your original approach, but I feel like for a large enough grid (and deep enough profile), the array of zeros could have significant overhead for both memory and runtime.

Reply:
Of course now that makes the `get_layer` call conditional as well, but I think that’s worth saving the extra calculation. You’d also need another test case for the case where you’re not passing in w. Thoughts, @jthielen ?

Reply:
That sounds good to reduce the overhead of the zeros approach. I have made the calls conditional now in my most recent commit, and added tests to cover both situations. Let me know if that is what you were thinking.